PaymentSystemPrototype/Services/AuthService.cs

```
@@ -16,19 +17,20 @@ public AuthService(IUserOperationsService userOperationsService)
         _userOperationsService = userOperationsService;
     }
     
-    public async Task<HttpStatusCode> LogInAsync(string userEmail, string userPassword, HttpContext httpContext)
+    public async Task<bool> LogInAsync(string userEmail, string userPassword, HttpContext httpContext)
     {
-        if (_userOperationsService.FindByEmail(userEmail) == null)
+        if (await _userOperationsService.FindByEmailAsync(userEmail) == null)
         {
-            return HttpStatusCode.NotFound;
+            throw new UserNotFoundException();
         }
-        var account = await _userOperationsService.CheckLoginInfo(userEmail, userPassword);
+        var account = await _userOperationsService.CheckLoginInfoAsync(userEmail, userPassword);
         if (account != null)
```

С точки зрения организации кода лучше избегать вложенности. То есть, большие куски кода по возможности лучше не сдвигать вправо.

Также хорошо использовать принцип fail fast: если ошибка, то прекращаем сразу.

В этом случае оба принципа достигаются инвертированием if.